### PR TITLE
[H02] Migrate to the latest version of Compound oracle 

### PR DESCRIPTION
**Pull request by mshrieve**
Let me know what you think!

### CODE REVIEW

File: test/oracle.test.ts [hunk 1]
```diff
@@ -174,22 +174,24 @@ contract('Oracle.sol', ([owner, random, ...tokens]) => {
...
       // update compound oracle price for USDC
-      const usdcPrice = '2348189545860141000000000000';
-      await compoundOracle.updatePrice(usdcPrice); // ? wei per USDC
+      // in wei
+      const usdcPrice = '2348189545860141';
+      await compoundOracle.updateUnderlyingPriceInWei(cUSDC.address, usdcPrice); // ? wei per USDC
       // get USDC price
       const price = await oracle.getPrice(usdc.address);
+      // expect the same?
       assert.equal(price.toString(), '2348189545860141');
```
Comment (mshrieve) [inline test/oracle.test.ts:195]: I changed the decimals here on the price. I wasn't sure?

**Comment by mshrieve** [comment]
this is looking pretty good now imo. lmk what you all think :)

File: contracts/Oracle.sol [hunk 2]
```diff
@@ -72,10 +72,19 @@ contract Oracle is Ownable {
...
+        return priceOracle.price("BTC");
+    }
+
+    /**
+     * @dev get BTC price in USD
+     * @return Price in USD with 6 decimals.
+     */
+    function getETHPrice() external view returns (uint256) {
+        return priceOracle.price("ETH");
     }
```
Comment (antoncoding) [inline contracts/Oracle.sol:88]: Why making btc different from other assets? 

Can't we use `getPrice(_asset)` as the main interface as well?

**Review by antoncoding — CHANGES_REQUESTED** [review]
As long as we're updating the oracle, I think it would be better to 
1. remove all the hardcoded address in the constructor.
2. let the admin add asset => pricer pairs. (something like `setAssetPricer(_asset, _pricer)`). So we can update the oracle used for a specific asset in the future. For now we can link all the pricer to the new compound oracle.

File: contracts/interfaces/CompoundOracleInterface.sol [hunk 3]
```diff
@@ -4,5 +4,7 @@ import "../packages/ERC20.sol";
 
 
 interface CompoundOracleInterface {
-    function getUnderlyingPrice(IERC20 cToken) external view returns (uint256);
+    function getUnderlyingPrice(address cToken) external view returns (uint256);
+
+    function price(string calldata symbol) external view returns (uint256);
```
[the remark this answers is not in the document]
Comment (antoncoding) [reply contracts/interfaces/CompoundOracleInterface.sol:9]: Okay this was the old interface, better update the contract address at the top of the file (`CompoundOracleInterface.sol`) (or delete it)

File: contracts/Oracle.sol [hunk 4]
```diff
@@ -84,40 +93,59 @@ contract Oracle is Ownable {
      * For other assets: ex: USDC: return 2349016936412111
      *  => 1 USDC = 2349016936412111 wei
      *  => 1 ETH = 1e18 / 2349016936412111 USDC = 425.71 USDC
+     * @param asset The address of the token.
+     * @return The price in wei.
      */
+
     function getPrice(address asset) external view returns (uint256) {
         if (asset == address(0)) {
             return (10**18);
```
Comment (haythem96) [inline contracts/Oracle.sol:102]: We are not using ETH anymore, right ? can we remove this ? // cc @aparnakr @antoncoding

File: contracts/Oracle.sol (hunk 2 above)
Comment (mshrieve) [reply contracts/Oracle.sol:88]: getPrice takes an address of a token

File: contracts/Oracle.sol (hunk 4 above)
Comment (mshrieve) [reply contracts/Oracle.sol:102]: this is a very important feature which tells you how many wei are in 1 eth

Comment (mshrieve) [reply contracts/Oracle.sol:102]: im just kidding it should be fine to remove

Comment (mshrieve) [reply contracts/Oracle.sol:102]: actually... anyone else have thoughts on this?

Comment (antoncoding) [reply contracts/Oracle.sol:102]: we still need it in case we want to use ETH as collateral. (that part is not removed right?)

Comment (mshrieve) [reply contracts/Oracle.sol:102]: im pretty sure only OptionsContractever calls `oracle.getPrice`. and the `getPrice` of OptionsContract looks like this:

```   
function getPrice(address asset) internal view returns (uint256) {
  if (asset == address(0)) {
    return (10**18);
  } else {
    return oracle.getPrice(asset);
  } 
}

Comment (mshrieve) [reply contracts/Oracle.sol:102]: re: my stupid joke, maybe that we dont need the oracle to fetch price of eth in wei.
but maybe it makes sense to put it there as an all-in-one pricing module.
lmk if yall have thoughts

Comment (antoncoding) [reply contracts/Oracle.sol:102]: ohh now I see it. Definitely remove it 😂

**Review by aparnakr — APPROVED** [review]

File: contracts/Oracle.sol (hunk 4 above)
Comment (haythem96) [reply contracts/Oracle.sol:102]: @aparnakr thoughts ?

Comment (haythem96) [reply contracts/Oracle.sol:102]: @antoncoding we removed ETH from V1, right ?

Comment (haythem96) [reply contracts/Oracle.sol:102]: Oh okay I think it is better to remove the check if address equal to zero in the OptionContract.sol

Comment (mshrieve) [reply contracts/Oracle.sol:102]: ok cool so i will leave it in the oracle 👍🏻

File: contracts/Oracle.sol (hunk 2 above)
Comment (haythem96) [reply contracts/Oracle.sol:88]: Please resolve this comment if there is no other open question.

**Comment by haythem96** [comment]
If time permit, I suggest removing all those variables as they are not needed: (the variables + their getters and setters functions)
```
        cEth = 0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5;
        cBat = 0x6C8c6b02E7b2BE14d4fA6022Dfd6d75921D90E4E;
        cDai = 0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643;
        cRep = 0x158079Ee67Fce2f58472A96584A73C7Ab9AC95c1;
        cUsdc = 0x39AA39c021dfbaE8faC545936693aC917d5E7563;
        cWbtc = 0xC11b1268C1A384e55C48c2391d8d480264A3A7F4;
        cZrx = 0xB3319f5D18Bc0D84dD1b4825Dcde5d5f7266d407;

        bat = 0x0D8775F648430679A709E98d2b0Cb6250d2887EF;
        dai = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
        rep = 0x1985365e9f78359a9B6AD760e32412f4a445E862;
        usdc = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
        wbtc = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
        zrx = 0xE41d2489571d322189246DaFA5ebDe1F4699F498;
```

and have them hard-coded in the constructor.

e.g:
instead of `isCtoken[cBat] = true;` => `isCtoken[0x6C8c6b02E7b2BE14d4fA6022Dfd6d75921D90E4E] = true;`

File: contracts/Oracle.sol (hunk 2 above)
Comment (mshrieve) [reply contracts/Oracle.sol]: this have both been removed.

**Comment by mshrieve** [comment]
ok. ive  incorporated all the changes we discussed!

**Comment by antoncoding** [comment]
> If time permit, I suggest removing all those variables as they are not needed: (the variables + their getters and setters functions)
> 
> ```
>         cEth = 0x4Ddc2D193948926D02f9B1fE9e1daa0718270ED5;
>         cBat = 0x6C8c6b02E7b2BE14d4fA6022Dfd6d75921D90E4E;
>         cDai = 0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643;
>         cRep = 0x158079Ee67Fce2f58472A96584A73C7Ab9AC95c1;
>         cUsdc = 0x39AA39c021dfbaE8faC545936693aC917d5E7563;
>         cWbtc = 0xC11b1268C1A384e55C48c2391d8d480264A3A7F4;
>         cZrx = 0xB3319f5D18Bc0D84dD1b4825Dcde5d5f7266d407;
> 
>         bat = 0x0D8775F648430679A709E98d2b0Cb6250d2887EF;
>         dai = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
>         rep = 0x1985365e9f78359a9B6AD760e32412f4a445E862;
>         usdc = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
>         wbtc = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
>         zrx = 0xE41d2489571d322189246DaFA5ebDe1F4699F498;
> ```
> 
> and have them hard-coded in the constructor.
> 
> e.g:
> instead of `isCtoken[cBat] = true;` => `isCtoken[0x6C8c6b02E7b2BE14d4fA6022Dfd6d75921D90E4E] = true;`

This is already an Ownable contract, can we just have a ownerOnly function that can set isCtoken = true, and remove all the hardcoded addresses in the constructor? 

```javascript
function setIsCToken(address _token, bool _isCToken) onlyOwer external {
   isCtoken[_token] = _isCToken;
}
```

File: contracts/Oracle.sol [hunk 5]
```diff
@@ -38,142 +24,110 @@ contract Oracle is Ownable {
...
-        return zrx;
-    }
-
     /**
-     * Asset Setters
+     * Asset Getters
      */
+    function iscEth(address asset) external view returns (bool) {
+        return asset == cEth;
+    }
```
Comment (antoncoding) [inline contracts/Oracle.sol:126]: we don't need this as well

Comment (mshrieve) [reply contracts/Oracle.sol]: removed

**Comment by mshrieve** [comment]
re: anton's comment, I removed hard-coded address

**Review by antoncoding — APPROVED** [review]

**Review by haythem96 — APPROVED** [review]
LGTM

**Pull request merged**